src/tiled/propertybrowser.cpp

```diff
@@ -551,16 +552,13 @@ void PropertyBrowser::propertiesChanged(Object *object)
         updateCustomProperties();
 }
 
-void PropertyBrowser::componentAdded(Object *, const QString &)
+void PropertyBrowser::componentAdded(QList<Object *> objects, const QString &)
 {
     addComponents();
 }
 
-void PropertyBrowser::removeComponent(Object *object, const QString &componentName)
+void PropertyBrowser::removeComponent(QList<Object *> objects, const QString &componentName)
 {
-    if (object != mObject)
-        return;
```

Comment:
Hmm, we can't just remove this check. Now `PropertyBrowser::removeComponent` removes the components from the properties view regardless of which objects are selected. This can be wrong especially in case of "redo" with different objects selected. I'll look into fixing that.